core/storage/src/relational/insert.rs

```diff
@@ -27,335 +26,683 @@ macro_rules! save_batch_slice {
 }
 
 impl RelationalStorage {
-    #[tracing_async]
     pub(crate) async fn insert_block_table(
         &self,
-        _ctx: Context,
         block_view: &BlockView,
-        tx: &mut RBatisTxExecutor<'_>,
+        tx: &mut Transaction<'_, Any>,
     ) -> Result<()> {
-        let block_hash = to_rb_bytes(&block_view.hash().raw_data());
+        // insert mercury_block and mercury_canonical_chain
+        bulk_insert_blocks(&[block_view.to_owned()], tx).await?;
 
-        tx.save(&BlockTable::from(block_view), &[]).await?;
-        tx.save(&SyncStatus::new(block_view.number()), &[]).await?;
-        tx.save(
-            &CanonicalChainTable::new(block_view.number(), block_hash),
-            &[],
+        // insert mercury_sync_status
+        SQLXPool::new_query(
+            r#"INSERT INTO mercury_sync_status(block_number)
+            VALUES ($1)"#,
         )
-        .await?;
-
-        Ok(())
+        .bind(i32::try_from(block_view.number())?)
+        .execute(&mut *tx)
+        .await
+        .map(|_| ())
+        .map_err(Into::into)
```

Comment:
you may use the question mark sugar:
```suggestion
        .await?;
        Ok(())
```